### PR TITLE
[HDT-16] Feat: Add `Heading block` component

### PR DESCRIPTION
**Pull request by RBoraurito**
## Assets

| before | after |
| ------ | ----- |
|        |       |

## Before Merge Checklist

The following steps were checked before merging a PR

- [x] checked and fixed code style issues
- [x] checked and removed unnecessary/junky code
- [x] manually tested introduced solution
- [ ] added needed tests to automate testing this code in the future

### CODE REVIEW

File: src/components/modules/heading-block/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  $align = 'center',
+  $centered = false,
+  title,
+  text,
+  badgeTitle,
+  badgeAdditionalText,
+  badgeIcon,
+}: HeadingBlockProps) => {
+  const formattedTitle = title.split('`').map(segment => {
+    const isGradient = segment.includes('*');
```
Comment (nephlin7) [inline src/components/modules/heading-block/index.tsx]: Could we name it `isHighlighted` instead?

File: src/components/modules/heading-block/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { StyledHeadingBlockWrapper, StyledHeadingBlockTitle } from './heading-block.styles';
+import type { HeadingBlockProps } from './heading-block.types';
+
+import { Badge } from '@/components/elements/badge';
+import { Text } from '@/components/elements/text';
+
+/**
+ * @param title: This string can have text gradient on the segments that are between \`\*content\*\`
```
Comment (nephlin7) [inline src/components/modules/heading-block/index.tsx]:
```suggestion
 * @param title: This string can have highlighted text on the segments that are between \`\*content\*\`
```

File: src/components/modules/heading-block/heading-block.types.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import type { BadgeProps } from '@/components/elements/badge/badge.types';
+
+export interface StyledHeadingBlockProps {
+  $align: CanvasTextAlign;
+  $centered?: boolean;
```
Comment (nephlin7) [inline src/components/modules/heading-block/heading-block.types.ts]: Just to differentiate the purpose of the alignment props.
 
``` suggestion
  $isContentCentered?: boolean;
```

**Review by nephlin7 — APPROVED** [review]
Great job @RBoraurito 🙌🏽 . I just left some small suggestions. Take a look before merging, please.

File: src/components/modules/heading-block/index.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  return (
+    <StyledHeadingBlockWrapper $align={$align} $centered={$centered}>
+      {badgeTitle && (
+        <Badge variant="secondary" additionalText={badgeAdditionalText} icon={badgeIcon}>
+          {badgeTitle}
+        </Badge>
+      )}
+      <StyledHeadingBlockTitle fontSize="l" textAlign={$align}>
```
Comment (nephlin7) [inline src/components/modules/heading-block/index.tsx]: The default color should be the textGradient
```suggestion
      <StyledHeadingBlockTitle fontSize="l" textAlign={$align}  color="textGradient">
```

Comment (RBoraurito) [reply src/components/modules/heading-block/index.tsx]: Now that I look at it, is not gradient xd, it is just cinnabar red

**Pull request merged**